### ISSUE TITLE
Segmentation fault

### ISSUE THREAD

**Issue by watermoon123** [jkxing/DROT#4]
running python experiments/evaluation/optim.py. I got segmentation fault as follows:

Setting up [LPIPS] perceptual loss: trunk [alex], v[0.1], spatial [off]
.../site-packages/torchvision/models/_utils.py:209: UserWarning: The parameter 'pretrained' is deprecated since 0.13 and will be removed in 0.15, please use 'weights' instead.
  f"The parameter '{pretrained_param}' is deprecated since 0.13 and will be removed in 0.15, "
.../site-packages/torchvision/models/_utils.py:223: UserWarning: Arguments other than a weight enum or `None` for 'weights' are deprecated since 0.13 and will be removed in 0.15. The current behavior is equivalent to passing `weights=AlexNet_Weights.IMAGENET1K_V1`. You can also use `weights=AlexNet_Weights.DEFAULT` to get the most up-to-date weights.
  warnings.warn(msg)
Loading model from: .../python3.7/site-packages/lpips/weights/v0.1/alex.pth
[KeOps] Compiling cuda jit compiler engine ... OK
[pyKeOps] Compiling nvrtc binder for python ... OK
no texture save
.../python3.7/site-packages/torch/functional.py:478: UserWarning: torch.meshgrid: in an upcoming release, it will be required to pass the indexing argument. (Triggered internally at  /opt/conda/conda-bld/pytorch_1656352464346/work/aten/src/ATen/native/TensorShape.cpp:2894.)
  return _VF.meshgrid(tensors, **kwargs)  # type: ignore[attr-defined]
no texture save
  0%|                                                  | 0/1000 [00:00<?, ?it/s][KeOps] Generating code for formula Max_SumShiftExpWeight_Reduction(Concat(Var(2,1,1)-Var(3,1,2)*(((Var(0,5,0)-Var(1,5,1))|(Var(0,5,0)-Var(1,5,1)))/2),1),0) ... Segmentation fault (core dumped)

Here is my conda list shows:
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                        main    defaults
_openmp_mutex             5.1                       1_gnu    defaults
absl-py                   2.0.0                    pypi_0    pypi
blas                      1.0                         mkl    defaults
brotlipy                  0.7.0           py38h27cfd23_1003    defaults
bzip2                     1.0.8                h7b6447c_0    defaults
ca-certificates           2023.08.22           h06a4308_0    defaults
cachetools                5.3.1                    pypi_0    pypi
certifi                   2023.7.22        py38h06a4308_0    defaults
cffi                      1.15.1           py38h5eee18b_3    defaults
charset-normalizer        2.0.4              pyhd3eb1b0_0    defaults
colorama                  0.4.6              pyhd8ed1ab_0    conda-forge
contourpy                 1.1.1                    pypi_0    pypi
cryptography              41.0.3           py38hdda0065_0    defaults
cudatoolkit               11.3.1               h2bc3f7f_2    defaults
cycler                    0.12.1                   pypi_0    pypi
ffmpeg                    4.3                  hf484d3e_0    pytorch
fonttools                 4.43.1                   pypi_0    pypi
freetype                  2.12.1               h4a9f257_0    defaults
fvcore                    0.1.5.post20210915            py38    fvcore
geomloss                  0.2.6                    pypi_0    pypi
giflib                    5.2.1                h5eee18b_3    defaults
gmp                       6.2.1                h295c915_3    defaults
gnutls                    3.6.15               he1e5248_0    defaults
google-auth               2.23.3                   pypi_0    pypi
google-auth-oauthlib      1.0.0                    pypi_0    pypi
grpcio                    1.59.0                   pypi_0    pypi
idna                      3.4              py38h06a4308_0    defaults
ignite                    0.4.13                     py_0    pytorch
imageio                   2.31.5                   pypi_0    pypi
importlib-metadata        6.8.0                    pypi_0    pypi
importlib-resources       6.1.0                    pypi_0    pypi
intel-openmp              2023.1.0         hdb19cb5_46305    defaults
iopath                    0.1.9                      py38    iopath
jpeg                      9e                   h5eee18b_1    defaults
keopscore                 2.1.2                    pypi_0    pypi
kiwisolver                1.4.5                    pypi_0    pypi
lame                      3.100                h7b6447c_0    defaults
lcms2                     2.12                 h3be6417_0    defaults
ld_impl_linux-64          2.38                 h1181459_1    defaults
lerc                      3.0                  h295c915_0    defaults
libdeflate                1.17                 h5eee18b_1    defaults
libffi                    3.4.4                h6a678d5_0    defaults
libgcc-ng                 11.2.0               h1234567_1    defaults
libgomp                   11.2.0               h1234567_1    defaults
libiconv                  1.16                 h7f8727e_2    defaults
libidn2                   2.3.4                h5eee18b_0    defaults
libpng                    1.6.39               h5eee18b_0    defaults
libstdcxx-ng              11.2.0               h1234567_1    defaults
libtasn1                  4.19.0               h5eee18b_0    defaults
libtiff                   4.5.1                h6a678d5_0    defaults
libunistring              0.9.10               h27cfd23_0    defaults
libwebp                   1.3.2                h11a3e52_0    defaults
libwebp-base              1.3.2                h5eee18b_0    defaults
lpips                     0.1.4                    pypi_0    pypi
lz4-c                     1.9.4                h6a678d5_0    defaults
markdown                  3.5                      pypi_0    pypi
markupsafe                2.1.3                    pypi_0    pypi
matplotlib                3.7.3                    pypi_0    pypi
mkl                       2023.1.0         h213fc3f_46343    defaults
mkl-service               2.4.0            py38h5eee18b_1    defaults
mkl_fft                   1.3.8            py38h5eee18b_0    defaults
mkl_random                1.2.4            py38hdb19cb5_0    defaults
ncurses                   6.4                  h6a678d5_0    defaults
nettle                    3.7.3                hbbd107a_1    defaults
ninja                     1.11.1.1                 pypi_0    pypi
numpy                     1.24.3           py38hf6e8229_1    defaults
numpy-base                1.24.3           py38h060ed82_1    defaults
nvdiffrast                0.3.1                    pypi_0    pypi
nvidiacub                 1.10.0                        0    bottler
oauthlib                  3.2.2                    pypi_0    pypi
opencv-python             4.8.1.78                 pypi_0    pypi
openh264                  2.1.1                h4ff587b_0    defaults
openjpeg                  2.4.0                h3ad879b_0    defaults
openssl                   3.0.11               h7f8727e_2    defaults
packaging                 23.2                     pypi_0    pypi
pillow                    10.0.1           py38ha6cbd5a_0    defaults
pip                       23.3             py38h06a4308_0    defaults
portalocker               1.4.0                      py_0    conda-forge
protobuf                  4.24.4                   pypi_0    pypi
pyasn1                    0.5.0                    pypi_0    pypi
pyasn1-modules            0.3.0                    pypi_0    pypi
pybind11                  2.11.1                   pypi_0    pypi
pycparser                 2.21               pyhd3eb1b0_0    defaults
pyglm                     2.7.0                    pypi_0    pypi
pykeops                   2.1.2                    pypi_0    pypi
pyopenssl                 23.2.0           py38h06a4308_0    defaults
pyparsing                 3.1.1                    pypi_0    pypi
pysocks                   1.7.1            py38h06a4308_0    defaults
python                    3.8.18               h955ad1f_0    defaults
python-dateutil           2.8.2                    pypi_0    pypi
pytorch                   1.12.0          py3.8_cuda11.3_cudnn8.3.2_0    pytorch
pytorch-mutex             1.0                        cuda    pytorch
pytorch3d                 0.7.0                    pypi_0    pypi
pyyaml                    5.1.2            py38h516909a_1    conda-forge
readline                  8.2                  h5eee18b_0    defaults
requests                  2.31.0           py38h06a4308_0    defaults
requests-oauthlib         1.3.1                    pypi_0    pypi
rsa                       4.9                      pypi_0    pypi
scipy                     1.10.1                   pypi_0    pypi
setuptools                68.0.0           py38h06a4308_0    defaults
six                       1.16.0                   pypi_0    pypi
sqlite                    3.41.2               h5eee18b_0    defaults
tabulate                  0.9.0              pyhd8ed1ab_1    conda-forge
tbb                       2021.8.0             hdb19cb5_0    defaults
tensorboard               2.14.0                   pypi_0    pypi
tensorboard-data-server   0.7.1                    pypi_0    pypi
termcolor                 2.3.0              pyhd8ed1ab_0    conda-forge
tk                        8.6.12               h1ccaba5_0    defaults
torchaudio                0.12.0               py38_cu113    pytorch
torchvision               0.13.0               py38_cu113    pytorch
tqdm                      4.66.1             pyhd8ed1ab_0    conda-forge
typing_extensions         4.7.1            py38h06a4308_0    defaults
urllib3                   1.26.16          py38h06a4308_0    defaults
werkzeug                  3.0.0                    pypi_0    pypi
wheel                     0.41.2           py38h06a4308_0    defaults
xmltodict                 0.13.0                   pypi_0    pypi
xz                        5.4.2                h5eee18b_0    defaults
yacs                      0.1.8              pyhd8ed1ab_0    conda-forge
yaml                      0.2.5                h7f98852_2    conda-forge
zipp                      3.17.0                   pypi_0    pypi
zlib                      1.2.13               h5eee18b_0    defaults
zstd                      1.5.5                hc292b87_0    defaults

**Comment by jkxing** [comment]
The error seems like raised from geomloss library. Have you pass the tests provided by geom loss?
